plugins/mac-capture/mac-sck-video-capture.m

```diff
@@ -4,14 +4,17 @@
 API_AVAILABLE(macos(12.5)) static void destroy_screen_stream(struct screen_capture *sc)
 {
     if (sc->disp && !sc->capture_failed) {
+        os_event_t *disp_finished;
```

Comment:
Same question as above?

Would it be possible to set the pointers inside the struct to NULL after copying them into local variables which are then used inside the completion handler to release the Cocoa objects and thus avoid the synchronisation entirely?